mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -2525,7 +2529,7 @@ struct Orderbook {
     topics_subscribed_to: HashMap<String, OrderbookRequestingState>,
     /// MemoryDB instance to store Patricia Tries data
     memory_db: MemoryDB<Blake2Hasher64>,
-    my_p2p_pubkeys: HashSet<String>,
+    my_zhtlc_orders_pubkeys: HashSet<String>,
```

Comment:
This will be used for any privacy coin, it doesn't have to be `zhtlc`. Please change the name to reflect this or just roll it back to what it was.

Reply:
as I understand [zhtlc is not a coin](https://developers.komodoplatform.com/basic-docs/atomicdex-api-20-dev/zhtlc_coins.html#task-enable-z-coin-init). That is an Komodo's umbrella protocol to operate on all private coins aka "any privacy coin"  [according to this comment](https://github.com/KomodoPlatform/atomicDEX-API/blob/1015-best-orders-is_mine/mm2src/mm2_main/src/lp_ordermatch.rs#L362C1-L364). Correct me if I'm wrong

cc: @laruh

Reply:
There are other coins that can have privacy features that is not under the zhtlc protocol (e.g. lightning, although the privacy part is not our focus now but can be added in the future). To implement private orders/swaps for these we would need to have a different `p2p_pubkey` for every order/swap so it's better to name this something other than `my_zhtlc_orders_pubkeys`

Reply:
@rozhkovdmitrii thanks for the ping!
ZHTLC coins are specific coins like [ARRR or ZOMBIE](https://developers.komodoplatform.com/basic-docs/atomicdex-api-20-dev/zhtlc_coins.html#zhtlc-coin-methods) which utilize Zcash protocol with its **shielded transactions** feature. You can find coins which have ZHTL protocol type in [coins](https://github.com/KomodoPlatform/coins/blob/master/coins) file.

As @shamardy said Zcash protocol has its features but it doesn't mean that other coins (not ZHTLC protocol type) cant have some common implementations.

@shamardy lets just change the comment above `is_mine_order` to this

```
// Some coins, for example ZHTLC, have privacy features like random keypair to sign P2P messages per every order.  
// So, each order of such coin has unique «pubkey» field that doesn’t match node persistent pubkey derived from passphrase.  
// We can compare pubkeys from maker_orders and from asks or bids, to find our order.
```

Reply:
[done](https://github.com/KomodoPlatform/atomicDEX-API/pull/1849/commits/263aa981e0d01005cff6ae74cdc61c2e6feed88e)